trappy/ftrace.py

```diff
@@ -168,28 +168,21 @@ class definitions list.  Otherwise, register a class to parse
     def __populate_data(self, fin, cls_for_unique_word, window, abs_window):
         """Append to trace data from a txt trace"""
 
-        def contains_unique_word(line, unique_words=cls_for_unique_word.keys()):
-            for unique_word in unique_words:
-                if unique_word in line:
-                    return True
-            return False
-
         actual_trace = itertools.dropwhile(self.trace_hasnt_started(), fin)
         actual_trace = itertools.takewhile(self.trace_hasnt_finished(),
                                            actual_trace)
 
         for line in actual_trace:
-            if not contains_unique_word(line):
-                self.lines += 1
-                continue
+            trace_class = None
```

Comment:
While we are at that, maybe we can cleanup this further by moving the following block into a local function, which replaces the one you remove above, e.g.

```python
def get_trace_class(line):
    trace_class = None
    for unique_word, cls in cls_for_unique_word.iteritems():
        if unique_word in line:
            trace_class = cls
            if not cls.fallback:
                return trace_class
    return trace_class
```

With that, the beginning of the parsing loop should be more simple "to parse":

```python
for line in actual_trace:
    trace_class = self.get_trace_class(line)
    if not trace_class:
        # Line not matching any of the enabled/registered attention words
        # still we account for the skipped trace line
        self.line += 1
        continue
    line = line[:-1]
   #...
```